src/appleseed/renderer/kernel/shading/shadingpoint.cpp

```diff
@@ -371,69 +373,98 @@ void ShadingPoint::refine_and_offset() const
 
     // Compute the location of the intersection point in assembly instance space.
     ShadingRay::RayType asm_inst_ray = m_assembly_instance_transform.to_local(m_ray);
-    asm_inst_ray.m_org += asm_inst_ray.m_tmax * asm_inst_ray.m_dir;
 
     switch (m_primitive_type)
     {
       case PrimitiveTriangle:
         {
+            // Offset the ray origin to the hit point.
+            asm_inst_ray.m_org += asm_inst_ray.m_tmax * asm_inst_ray.m_dir;
+
+            const TriangleSupportPlaneType& triangle_support_plane = m_triangle_support_plane;
```

Comment:
I added this variable because of this 
![image](https://user-images.githubusercontent.com/4656466/67634891-7792d980-f8c1-11e9-8d82-68a9f8bd6137.png)

Reply:
Got it 

```
const auto intersection_handling = [&triangle_support_plane = m_triangle_support_plane](const Vector3d& p, const Vector3d& n) {
                return triangle_support_plane.intersect(p, n);
            };
```

Reply:
That's C++14 which we aren't using yet.

If you capture `this`, it will work nicely and efficiently.